runtime/io/common/src/aws/smithy/kotlin/runtime/io/SdkManaged.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.io
+
+import aws.smithy.kotlin.runtime.util.InternalApi
+import kotlinx.atomicfu.locks.SynchronizedObject
+import kotlinx.atomicfu.locks.synchronized
+
+/**
+ * Interface that the lifecycle of some resource is managed by the SDK at runtime.
+ */
+@InternalApi
+public interface SdkManaged {
```

Comment:
**Question**: Would you say this belongs in `runtime-core` long term? (I understand why it is here right now).

Reply:
100%